playbooks/e2e.yaml

```diff
@@ -0,0 +1,81 @@
+  tasks:
+  - name: get ssh version
+    command: "ssh -V"
+  - name: Prepare environment
+    block:
+    - name: Get PR details
+      uri:
+        url: "https://api.github.com/repos/{{ zuul.project.name }}/pulls/{{ zuul.change }}"
+        body_format: json
+        headers:
+          Accept: application/vnd.github+json
+          X-GitHub-Api-Version: 2022-11-28
+      register: pull_request
+    - name: Set facts based on PR details
+      set_fact:
+        git_branch_name: "{{ pull_request.json.head.ref }}"
+        git_repository_url: "{{ pull_request.json.head.repo.clone_url }}"
+    - name: Create environment file
+      template:
+        src: "templates/environment.tfvars.j2"
+        dest: "{{ project_tf_dir }}/environments/environment-{{ cloud_provider }}.tfvars"
+  - name: Create k8s cluster and execute sonobouy {{ sonobouy_mode }} check
+    block:
+    - name: Create k8s cluster
+      command: "make create GITREFERENCE={{ git_branch_name }} GITREPO={{ git_repository_url }}"
+      args:
+        chdir: "{{ project_tf_dir }}"
+      environment:
+        TF_LOG: DEBUG
+    - name: Get kubeconfig of the workload k8s cluster
+      command: "make get-kubeconfig TESTCLUSTER={{ testcluster_name }}"
+      args:
+        chdir: "{{ project_tf_dir }}"
+    - name: Wait for all system pods in the workload k8s cluster to become ready
+      command: "kubectl wait -n kube-system --for=condition=Ready --timeout={{ wait_for_cluster }}s pod --all"
+      args:
+        chdir: "{{ project_tf_dir }}"
+      environment:
+        KUBECONFIG: "{{ testcluster_name }}.yaml.{{ cloud_provider }}"
+    - name: Execute sonobouy {{ sonobouy_mode }} check
+      command: "make check-{{ sonobouy_mode }}"
+      register: sonobouy_results
+      args:
+        chdir: "{{ project_tf_dir }}"
+    always:
+    - name: expose private
+      command: "cat .deploy.id_rsa.{{ cloud_provider }}"
+      args:
+        chdir: "{{ project_tf_dir }}"
+    - name: pause
+      pause:
+        seconds: 360
+    - name: Parse sonobouy results
+      set_fact:
+        sonobouy_results_parsed: "{{ sonobouy_results.stdout | regex_search('=== Collecting results ===[\\S\\s]*') }}"
+      when: sonobouy_results is defined
+      ignore_errors: true  # ignore when this task failed as it is nice to have, and we should ensure that fullclean is reached
+    - name: Insert parsed sonobouy results to the warning message that will be appended to the comment zuul leaves on the PR
+      zuul_return:
+        data:
+          zuul:
+            warnings:
+              - "<details>\n  <summary><b>Sonobouy results</b></summary>\n{{ sonobouy_results_parsed }}\n</details>"
+      when: sonobouy_results_parsed is defined and sonobouy_results_parsed | length > 0
+      ignore_errors: true  # ignore when this task failed as it is nice to have, and we should ensure that fullclean is reached
+    - name: Cleanup - fullclean
```

Comment:
I think this should be part of the clean play to ensure that you always cleanup.